framework/src/partitioner/LibmeshPartitioner.C

```diff
@@ -82,3 +82,54 @@ LibmeshPartitioner::getPartitioner()
 {
   return _partitioner;
 }
+
+UniquePtr<Partitioner>
+LibmeshPartitioner::clone() const
+{
+//  UniquePtr<Partitioner> cloned_partitioner;
+  switch (_partitioner_name)
+  {
+  case -2: // metis
```

Comment:
Well I did some looking because I'm the one who originally added the negative numbers in the enum. The logic looks a lot different then it did when I originally added it here: 4d94e8701ced7c819e6a5294ca9274df1e208696. I should have documented it then but I believe I was thinking that the negative numbers were automatically assigned while the non-negative numbers where the user-overrides. There may even be a comparison against non-negative somewhere.